### PR TITLE
feat(schema, api-client): Migrate project schemas and environment schemas along with their types to @keyshade/schema

### PR DESCRIPTION
**Pull request by muntaxir4**
## Description

**`@keyshade/schema`**
- Add schemas and types for `project` and `environment`.
- Add tests for `project` and  `environment` schemas.
- Export both schemas from `index` and their types from `index.types`
- Remove existing `project.ts` and `environment.ts`. Modify `project` schema to include new `CreateEnvironmentRequestSchema`.

**`@keyshade/api-client`**
- Remove existing `project.types` and `environment.types`.
- Change imports for project and environment types to `@keyshade/schema`. <br />

Related to #519

## Developer's checklist

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-check on my work

**If changes are made in the code:**

- [x] I have followed the [coding guidelines](https://google.github.io/styleguide/jsguide.html)
- [x] My changes in code generate no new warnings
- [ ] My changes are breaking another fix/feature of the project
- [x] I have added test cases to show that my feature works
- [ ] I have added relevant screenshots in my PR
- [x] There are no UI/UX issues

### Documentation Update

- [ ] This PR requires an update to the documentation at [docs.keyshade.xyz](https://docs.keyshade.xyz)
- [x] I have made the necessary updates to the documentation, or no documentation changes are required.

___

### **PR Type**
enhancement, tests

___

### **Description**
- Migrated environment and project schemas and types from `api-client` to `@keyshade/schema`.
- Added new zod schemas for environment and project in the `schema` package.
- Updated `api-client` to use the new schemas from `@keyshade/schema`.
- Removed old type definitions from `api-client`.
- Added comprehensive tests for the new environment and project schemas.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><details><summary>10 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>environment.ts</strong><dd><code>Update environment controller to use new schema imports</code>&nbsp; &nbsp; </dd></summary>
<hr>

packages/api-client/src/controllers/environment.ts

- Updated imports to use schemas from `@keyshade/schema`.

</details>

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/538/files#diff-c932cd28fbb3efdccfe97d49717c63add1fb832486cf5eab0280087035d116c4">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>project.ts</strong><dd><code>Update project controller to use new schema imports</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/api-client/src/controllers/project.ts

- Updated imports to use schemas from `@keyshade/schema`.

</details>

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/538/files#diff-6d7292205bf262ca8e8eeb87c9a55bdeba0702b79dd5d9e8607a84c686e428c7">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>environment.types.d.ts</strong><dd><code>Remove environment type definitions from api-client</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/api-client/src/types/environment.types.d.ts

- Removed environment type definitions.

</details>

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/538/files#diff-8c2ab40c20c0a6143d7b62328d86c7cce1182c98c17974fd8c5269ae84c60c19">+0/-55</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>project.types.d.ts</strong><dd><code>Remove project type definitions from api-client</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/api-client/src/types/project.types.d.ts

- Removed project type definitions.

</details>

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/538/files#diff-e273fb89642b7f7ce61310da649ed205e53e429c34acedb4bda8a41aa74d71de">+0/-87</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>environment.ts</strong><dd><code>Add environment schemas using zod</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/schema/src/environment/environment.ts

- Added environment schemas using zod.

</details>

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/538/files#diff-606537c9a7d4b1ed1bbfb50fa2347721f538c20ccdfcd44b540ce3311bbdfd32">+56/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>environment.types.ts</strong><dd><code>Add inferred types for environment schemas</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/schema/src/environment/environment.types.ts

- Added inferred types for environment schemas.

</details>

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/538/files#diff-3dce6e9f5a2ba7ada2fba42b51591d71963bd7d2a5106d1c1140f5f5bd5fe09c">+54/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>index.ts</strong><dd><code>Update index exports for environment and project</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/schema/src/index.ts

- Updated exports for environment and project schemas.

</details>

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/538/files#diff-5c37949968eac79dd5d1c0df1176a9073fd624d2be2742d3c7d718e318aa2cdc">+5/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>index.types.ts</strong><dd><code>Update index type exports for environment and project</code>&nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/schema/src/index.types.ts

- Updated type exports for environment and project.

</details>

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/538/files#diff-47f389a66e4515c1526d27bc56ae56141418a0a0419271b98077614f7778d3e5">+4/-11</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>project.ts</strong><dd><code>Add project schemas using zod</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/schema/src/project/project.ts

- Added project schemas using zod.

</details>

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/538/files#diff-61938810e6520694e1177a33777d51c68021bb0d1c0fbfed13be9ac94a52fd3d">+99/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>project.types.ts</strong><dd><code>Add inferred types for project schemas</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/schema/src/project/project.types.ts

- Added inferred types for project schemas.

</details>

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/538/files#diff-32c24230e55f2ac65d746847b26159c0972dcd49a9dca71a891ae05942e2aa11">+62/-0</a>&nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Tests</strong></td><td><details><summary>2 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>environment.spec.ts</strong><dd><code>Add tests for environment schemas</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/schema/tests/environment.spec.ts

- Added comprehensive tests for environment schemas.

</details>

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/538/files#diff-1b074cba90a8785d2bea5ccda58461d9cc5fdb052699ff8938f86ccfbe6f36cd">+312/-8</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>project.spec.ts</strong><dd><code>Add tests for project schemas</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/schema/tests/project.spec.ts

- Added comprehensive tests for project schemas.

</details>

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/538/files#diff-4d04c509cd4ab44ec5c87dc8bcc491d52f99f8fab4499aaa08459daa168f704f">+558/-21</a></td>

</tr>
</table></details></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**: Comment `/help "your question"` on any pull request to receive relevant information

### CODE REVIEW

**Comment by muntaxir4** [comment]
Why are the PR tests failing? I can not find any such issue

File: packages/schema/src/index.ts
Comment (rajdip-b) [inline packages/schema/src/index.ts:1]: Can you get rid of the extra spaces?

File: packages/schema/src/index.types.ts [hunk 1]
```diff
@@ -23,15 +17,14 @@ export type TUpdateApiKey = z.infer<typeof UpdateApiKeySchema>
 // Export types from auth.types.ts
 export * from './auth/auth.types'
 
-export type TCreateEnvironment = z.infer<typeof CreateEnvironmentSchema>
-export type TUpdateEnvironment = z.infer<typeof UpdateEnvironmentSchema>
+// Export types from environment.types.ts
```
Comment (rajdip-b) [inline packages/schema/src/index.types.ts:20]: Won't need these comments

**Review by rajdip-b — CHANGES_REQUESTED** [review]

File: packages/schema/src/index.ts
Comment (muntaxir4) [reply packages/schema/src/index.ts:1]: Sure, I found it more readable.

Comment (rajdip-b) [reply packages/schema/src/index.ts:1]: Ah

**Comment by rajdip-b** [comment]
If you can make the changes, I'll get this merged asap.

**Comment by muntaxir4** [comment]
> If you can make the changes, I'll get this merged asap.

Made the required changes.

**Review by rajdip-b — APPROVED** [review]

**Pull request merged**